x/cash/msg.go

```diff
@@ -91,47 +87,45 @@ func (f *FeeInfo) DefaultPayer(addr []byte) *FeeInfo {
 // Validate makes sure that this is sensible.
 // Note that fee must be present, even if 0
 func (f *FeeInfo) Validate() error {
+	var err error
 	if f == nil {
-		return errors.Wrap(errors.ErrInput, "nil fee info")
+		err = errors.Wrap(errors.ErrInput, "nil fee info")
 	}
 	fee := f.GetFees()
 	if fee == nil {
-		return errors.Wrap(errors.ErrAmount, "fees nil")
-	}
-	if err := fee.Validate(); err != nil {
-		return err
-	}
-	if !fee.IsNonNegative() {
-		return errors.Wrap(errors.ErrAmount, "negative fees")
+		err = errors.Append(err, errors.Wrap(errors.ErrAmount, "fees nil"))
+	} else {
+		err = errors.Append(err, fee.Validate())
+
+		if !fee.IsNonNegative() {
+			err = errors.Append(err, errors.Wrap(errors.ErrAmount, "negative fees"))
+		}
 	}
-	return weave.Address(f.Payer).Validate()
+
+	return errors.Append(err, weave.Address(f.Payer).Validate())
 }
 
 var _ weave.Msg = (*ConfigurationMsg)(nil)
 
 // Validate will skip any zero fields and validate the set ones
 // TODO: we should make it easier to reuse code with Configuration
 func (m *ConfigurationMsg) Validate() error {
+	var err error
 	c := m.Patch
 	if len(c.Owner) != 0 {
-		if err := c.Owner.Validate(); err != nil {
-			return errors.Wrap(err, "owner address")
-		}
+		err = c.Owner.Validate()
```

Comment:
Please don't remove Wraps. They were there for a good purpose.

We had tagged almost every validation message with a human-readable text to identify the field with errors to help debugging tests at least. Removing them now leads to the exact problem you were complaining about, having no field mappings.

Please leave them in now. We could replace it with `errors.Tag` instead of `errors.Wrap` in some future PR design possibly, but don't remove the info without an alternative.

Thinking about this, I think Appendf is a bad name, as we might want for Wrap or Tag...

AppendWrap, AppendWrapf, AppendTag as shortcuts???
Someone who likes simple APIs is gonna hate me for that proposal...